_posts/2020-3-29-managed-services-serverless.md

```diff
@@ -0,0 +1,187 @@
+
+# Demo 2: Comment Vibes App ([source](https://github.com/alexellis/comment-vibes))
+
+Already having Crossplane and OpenFaaS open and running, why not move on to another interesting application? Alex had put together the [Comment Vibes](https://github.com/alexellis/comment-vibes) app to show how OpenFaaS functions can be used with Github and a Postgres database to display emoji reactions in comments. The application was [originally demoed](https://www.youtube.com/watch?v=r4mEF8rtXWo) by Alex at an online meetup by provisioning a Digital Ocean database using the cloud provider console. However, we wanted to show how we could simplify this workflow using Crossplane.
+
+After setting up the Crossplane [GCP Provider](https://github.com/crossplane/provider-gcp), we created a `CloudSQLInstanceClass` with configuration for a Postgres database on GCP. The contents of the configuration looked like this:
+
+```yaml
+apiVersion: database.gcp.crossplane.io/v1beta1
+kind: CloudSQLInstanceClass
+metadata:
+  name: standard
+  labels:
+    app: comment-vibes
+specTemplate:
+  writeConnectionSecretsToNamespace: crossplane-system
+  forProvider:
+    databaseVersion: POSTGRES_11
+    region: us-west2
+    settings:
+      tier: db-custom-1-3840
+      dataDiskType: PD_SSD
+      dataDiskSizeGb: 10
+      ipConfiguration:
+        ipv4Enabled: true
+        authorizedNetworks:
+          - value: "0.0.0.0/0" # whitelist all IPs so we can access from local machine
+  providerRef:
+    name: gcp-provider
+  reclaimPolicy: Delete
+```
+
+As you can see, a Postgres database is a bit more complicated than an object storage bucket, but this level of detail is once again confined to the infrastructure operator, while the developer simply creates a `PostgreSQLInstance` claim that can be satisfied by the `CloudSQLInstanceClass`. The contents of the claim are significantly less verbose:
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  name: comment-vibes-db
+  namespace: openfaas-fn
+spec:
+  classSelector:
+    matchLabels:
+      app: comment-vibes
+  writeConnectionSecretToRef:
+    name: pgconn
+  engineVersion: "11"
+```
+
+After creating both the `CloudSQLInstanceClass` and the `PostgreSQLInstance` claim, GCP began to provision the database. This [was reflected](https://youtu.be/XphQgB87U-s?t=3190) in the state of the `CloudSQLInstance` object in our Kubernetes cluster, as it reported back `PENDING_CREATE`:
+
+![GCP Pending Create](/images/2020-03-30-managed-services-serverless/gcp-pending-create.png)
+
+After a short period of time, the Postgres database was fully provisioned and the connection information was propagated to the `openfaas-fn` namespace in the form of a Kubernetes `Secret`. We used the connection information to [setup our database and table schema](https://youtu.be/XphQgB87U-s?t=3628) prior to deploying our functions.
+
+```console
```

Comment:
Does that render correctly? I think we use ```bash everywhere else

Reply:
Used `console` here because Okteto post did. Seems to render fine